app/src/org/commcare/android/storage/FormSaveUtil.java

```diff
@@ -0,0 +1,44 @@
+package org.commcare.android.storage;
+
+import android.content.Context;
+import android.database.Cursor;
+import android.net.Uri;
+
+import org.commcare.dalvik.application.CommCareApplication;
+import org.commcare.dalvik.odk.provider.FormsProviderAPI;
+
+import java.util.Hashtable;
+
+/**
+ * @author Phillip Mates (pmates@dimagi.com).
+ */
+public class FormSaveUtil {
+
+    /**
+     * @param context used to query form instances through android's content provider framework
+     * @return A mapping from an installed form's namespace to its install path.
+     */
+    public static Hashtable<String, String> getNamespaceToFilePathMap(Context context) {
```

Comment:
Broke out this logic into a helper function from `DataPullTask` so that it can also be used in tests to load saved form instances.